generators/app/templates/hooks/pre-commit

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
```

Comment:
Definitely think this is valuable, but this file is so gross and I understand almost none of it. Is there any way we can curl this from somewhere or something so we don't need to maintain it or keep it in our repo?